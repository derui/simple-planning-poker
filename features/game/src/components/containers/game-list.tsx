import { hooks } from "../../hooks/facade.js";
import { GameListLayout } from "./game-list.layout.js";

export const GameList = function GameList(): JSX.Element {
  const { games, selectedGame, select, requestCreate } = hooks.useListGames();

  const handleSelect = (gameId: string) => {
    select(gameId);
  };

  const handleRequestCreate = () => {
    requestCreate();
  };

  return (
    <GameListLayout
      games={games}
      selectedGame={selectedGame?.id}
      onRequestCreate={handleRequestCreate}
      onSelect={handleSelect}
    />
  );
};
