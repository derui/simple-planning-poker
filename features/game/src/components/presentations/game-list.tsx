import { Variant } from "@spp/shared-color-variant";
import { Icon } from "@spp/ui-icon";
import { GameDto } from "../../atoms/dto.js";
import { GameListItem } from "./game-list-item.js";
import * as styles from "./game-list.css.js";

export interface Props {
  games: GameDto[];

  /**
   *  Callback when request to create a new game is received.
   */
  onCreate?: () => void;

  /**
   * Callback when a game is selected.
   */
  onSelect?: (id: string) => void;
}

export const GameList = function GameList({ games, onCreate, onSelect }: Props): JSX.Element {
  const items = games.map((v) => (
    <GameListItem key={v.id} gameId={v.id} name={v.name} selected={v.owned} onClick={() => onSelect?.(v.id)} />
  ));

  return (
    <ul className={styles.root}>
      {items}
      <li className={styles.plusContainer} onClick={onCreate}>
        <Icon.Plus size="m" variant={Variant.indigo} />
        Add Game
      </li>
    </ul>
  );
};
