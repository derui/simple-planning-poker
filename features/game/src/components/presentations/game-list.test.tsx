import { cleanup, render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import Sinon from "sinon";
import { afterEach, expect, test } from "vitest";
import { GameList } from "./game-list.js";

afterEach(cleanup);

test("should be able to render", async () => {
  const ret = render(<GameList games={[]} />);

  expect(ret.container).toMatchSnapshot();
});

test("should be render game", () => {
  const ret = render(<GameList games={[{ id: "id", name: "Game 1", owned: false }]} />);

  expect(ret.container).toMatchSnapshot();
});

test("should call callback after click Add Game", async () => {
  const callback = Sinon.fake();
  render(<GameList games={[]} onCreate={callback} />);

  await userEvent.click(screen.getByText("Add Game"));

  expect(callback.calledOnce).toBeTruthy();
});

test("should call callback after click a game", async () => {
  const callback = Sinon.fake();
  render(<GameList games={[{ id: "id", name: "Target", owned: false }]} onSelect={callback} />);

  await userEvent.click(screen.getByText("Target"));

  expect(callback.calledOnceWith("id")).toBeTruthy();
});
