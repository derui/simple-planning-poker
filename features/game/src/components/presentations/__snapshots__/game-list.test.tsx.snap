// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be able to render 1`] = `
<div>
  <ul
    class="game-list_root__17rv8c10"
  >
    <li
      class="game-list_plusContainer__17rv8c11 style_transition_all__192bcwg41"
    >
      <span
        class="plus_medium_indigo__12x1qz0h style_iconMedium__p90ga82 style_iconBase__p90ga80"
      />
      Add Game
    </li>
  </ul>
</div>
`;

exports[`should be render game 1`] = `
<div>
  <ul
    class="game-list_root__17rv8c10"
  >
    <li
      aria-selected="false"
      class="game-list-item_main__a15gnn0 style_transition_all__192bcwg41"
    >
      <div
        class="game-list-item_link__a15gnn1"
      >
        <span
          class="game-list-item_linkName__a15gnn2"
        >
          Game 1
        </span>
        <span
          class="game-list-item_selectMarkContainer__a15gnn3"
        >
          <span
            aria-label="mark"
            aria-selected="false"
            class="game-list-item_selectMark__a15gnn4"
          >
            <span
              class="check_medium_indigo__1oavie2h style_iconMedium__p90ga82 style_iconBase__p90ga80"
            />
          </span>
        </span>
      </div>
    </li>
    <li
      class="game-list_plusContainer__17rv8c11 style_transition_all__192bcwg41"
    >
      <span
        class="plus_medium_indigo__12x1qz0h style_iconMedium__p90ga82 style_iconBase__p90ga80"
      />
      Add Game
    </li>
  </ul>
</div>
`;
